### DISCUSSION TITLE
Very opinionated template style

### DISCUSSION THREAD

**Discussion by bardzh** [sveltejs/kit#6857] [Ideas]
I was thinking in terms of reducing scaffolding in the code, it might be a good idea to change `<script>` and `<style>` tags from 
```html
<script>
  /** @type {import('./$types').PageData} */
  export let name;
</script>
<style>
h1{
 font-size: 2rem;
}
</style>

<h1>{name}</h1>
``` 

to
```js
@code{  // or @script{}
 /** @type {import('./$types').PageData} */
  export let name;
}
@style{
  h1{
   font-size: 2rem;
  }
}
<h1>{name}</h1>
```
not sure what complications this could create but it was a thought in my head for a while

**Comment by gterras** [comment]
What would be the point of doing so?

**Comment by dominikg** [comment]
This would be a huge breaking change to svelte the language. And not an improvement either, just different. Not happening. Please also try to keep discussions in the kit repo about kit. If you want to discuss about svelte itself, there is https://svelte.dev/chat

**Comment by novotionyx** [comment]
I get the sentiment, but it doesn't really make sense. The whole reason a ton of people adore Svelte is due to its simplicity - this just complicates the syntax and the way you write components. It just makes logical sense that you write a script tag, a style tag then your markup - as that's what each components comprises of. Adding weird syntax to wrap what essentially compiles down to what we're currently writing anyways makes no sense.

**Reply by bardzh** [reply]
The idea of why i see resistance to this change. but also the reason we're using `script` and `style` is just traditions that we're stuck with. same as how we're stuck we `QWERTY` keyboard. i think things can be simpler that what they are. but again it's opinion.

**Reply by richarddavenport** [reply]
I don't know about you, but I certainly don't feel stuck with qwerty... it's what I use. Feel free to use something else.
As for the current svelte syntax, they're sticking with the web platform, which currently is HTML. It's not tradition, it's what it is.
I get trying to change things, but this is a change for sake of change and is grossly opinionated, to a point that I would suggest going and building your own svelte.

**Reply by aloker** [reply]
@script{} and @style{} look too similar to css rules like @media or @keyframes, imo. I'm fine with <script>/<style>.

**Reply by gterras** [reply]
>  but also the reason we're using script and style is just traditions

Lol or maybe also because it is part of the HTML spec, haven't you heard about web standards?  "Use the platform" is a pillar of Svelte and Sveltekit approach.

**Reply by gbsallery** [reply]
As a happy Colemak user (QWERTY really does suck), I still think that sticking to the HTML spec is a _way_ better idea than introducing esoteric and unfamiliar magic keywords.

**Comment by yousufiqbal** [comment]
Y..?
